From 396a6a9e49b16d4942290fc3d47bbbee221aa6da Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Sat, 25 Jan 2020 00:37:51 -0600
Subject: [PATCH] Register the Item field of more entities

This patches MC-169769. Without the fix, errors can
be thrown while converting these entities
---
 .../minecraft/server/DataConverterSchemaV1460.java   | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/main/java/net/minecraft/server/DataConverterSchemaV1460.java b/src/main/java/net/minecraft/server/DataConverterSchemaV1460.java
index f9a42ecec2..f7ce5ba55d 100644
--- a/src/main/java/net/minecraft/server/DataConverterSchemaV1460.java
+++ b/src/main/java/net/minecraft/server/DataConverterSchemaV1460.java
@@ -51,7 +51,7 @@ public class DataConverterSchemaV1460 extends DataConverterSchemaNamed {
             return DSL.optionalFields("Items", DSL.list(DataConverterTypes.ITEM_STACK.in(schema)), "SaddleItem", DataConverterTypes.ITEM_STACK.in(schema), DataConverterSchemaV100.a(schema));
         });
         schema.registerSimple(map, "minecraft:dragon_fireball");
-        schema.registerSimple(map, "minecraft:egg");
+        schema.register(map, "minecraft:egg", (s) -> DSL.optionalFields("Item", DataConverterTypes.ITEM_STACK.in(schema))); // EMC
         a(schema, map, "minecraft:elder_guardian");
         schema.registerSimple(map, "minecraft:ender_crystal");
         a(schema, map, "minecraft:ender_dragon");
@@ -59,10 +59,10 @@ public class DataConverterSchemaV1460 extends DataConverterSchemaNamed {
             return DSL.optionalFields("carriedBlockState", DataConverterTypes.BLOCK_STATE.in(schema), DataConverterSchemaV100.a(schema));
         });
         a(schema, map, "minecraft:endermite");
-        schema.registerSimple(map, "minecraft:ender_pearl");
+        schema.register(map, "minecraft:ender_pearl", (s) -> DSL.optionalFields("Item", DataConverterTypes.ITEM_STACK.in(schema))); // EMC
         schema.registerSimple(map, "minecraft:evocation_fangs");
         a(schema, map, "minecraft:evocation_illager");
-        schema.registerSimple(map, "minecraft:eye_of_ender_signal");
+        schema.register(map, "minecraft:eye_of_ender_signal", (s) -> DSL.optionalFields("Item", DataConverterTypes.ITEM_STACK.in(schema))); // EMC
         schema.register(map, "minecraft:falling_block", (s) -> {
             return DSL.optionalFields("BlockState", DataConverterTypes.BLOCK_STATE.in(schema), "TileEntityData", DataConverterTypes.BLOCK_ENTITY.in(schema));
         });
@@ -121,8 +121,8 @@ public class DataConverterSchemaV1460 extends DataConverterSchemaNamed {
             return DSL.optionalFields("SaddleItem", DataConverterTypes.ITEM_STACK.in(schema), DataConverterSchemaV100.a(schema));
         });
         a(schema, map, "minecraft:slime");
-        schema.registerSimple(map, "minecraft:small_fireball");
-        schema.registerSimple(map, "minecraft:snowball");
+        schema.register(map, "minecraft:small_fireball", (s) -> DSL.optionalFields("Item", DataConverterTypes.ITEM_STACK.in(schema))); // EMC
+        schema.register(map, "minecraft:snowball", (s) -> DSL.optionalFields("Item", DataConverterTypes.ITEM_STACK.in(schema))); // EMC
         a(schema, map, "minecraft:snowman");
         schema.register(map, "minecraft:spawner_minecart", (s) -> {
             return DSL.optionalFields("DisplayState", DataConverterTypes.BLOCK_STATE.in(schema), DataConverterTypes.UNTAGGED_SPAWNER.in(schema));
@@ -148,7 +148,7 @@ public class DataConverterSchemaV1460 extends DataConverterSchemaNamed {
         a(schema, map, "minecraft:wither_skeleton");
         schema.registerSimple(map, "minecraft:wither_skull");
         a(schema, map, "minecraft:wolf");
-        schema.registerSimple(map, "minecraft:xp_bottle");
+        schema.register(map, "minecraft:xp_bottle", (s) -> DSL.optionalFields("Item", DataConverterTypes.ITEM_STACK.in(schema))); // EMC
         schema.registerSimple(map, "minecraft:xp_orb");
         a(schema, map, "minecraft:zombie");
         schema.register(map, "minecraft:zombie_horse", (s) -> {
-- 
2.25.0

